server/src/api/v1.py

```diff
@@ -101,7 +100,33 @@ def has_attachments(data: dict) -> bool:
     )
 
 
-def job_builder(data):
+def check_token_permission(
+    auth_token: str, secret_key: str, priority: int, queue: str
+) -> bool:
+    """
+    Validates token received from client and checks if it can
+    push a job to the queue with the requested priority
+    """
+    if auth_token is None:
+        abort(401, "No authentication token specified")
+    else:
```

Comment:
Eliminate this entire `else` block. You did the right thing by checking the error condition first and aborting. But once you've done that, then we're on the happy path. No need for the `else` here :)